### PR TITLE
DYN-3177-Coverage-CoreNodeModelsWpf

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

I added three test methods for increasing code coverage in the CoreNodeModelsWpf\ConverterViewModel.cs file.
Also I modified the ConverterViewModel.cs due that is not consistent to the behavior in the SelectedToConversionSource property (ComboBox).

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@QilongTang 

### FYIs

@alfredo-pozo

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
Screenshot about code coverage for the ConverterViewModel class in 92%

image

The job DynamoSelfServe ran successfully for my forked repo.

https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/master/179/

File: src/Libraries/CoreNodeModelsWpf/ConverterViewModel.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ public List<ConversionUnit> SelectedToConversionSource
             get { return dynamoConvertModel.SelectedToConversionSource; }
             set
             {
-                dynamoConvertModel.SelectedFromConversionSource = value;             
+                dynamoConvertModel.SelectedToConversionSource = value;
```
Comment (QilongTang) [inline src/Libraries/CoreNodeModelsWpf/ConverterViewModel.cs:61]: 👍

File: test/DynamoCoreWpfTests/ConverterViewModelTests.cs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            var moduleRootFolder = Path.GetDirectoryName(assemblyPath);
+
+            var resolutionPaths = new[]
+            {
+                // The CoreNodeModelsWpf.dll needed is under "nodes" folder.
+                Path.Combine(moduleRootFolder, "nodes")
+            };
+
+            assemblyHelper = new AssemblyHelper(moduleRootFolder, resolutionPaths);
+            AppDomain.CurrentDomain.AssemblyResolve += assemblyHelper.ResolveAssembly;
```
Comment (QilongTang) [inline test/DynamoCoreWpfTests/ConverterViewModelTests.cs:60]: unsubscribe?

**Review by QilongTang — APPROVED** [review]
LGTM with one comment

File: test/DynamoCoreWpfTests/ConverterViewModelTests.cs (hunk 2 above)
Comment (RobertGlobant20) [reply test/DynamoCoreWpfTests/ConverterViewModelTests.cs:60]: @QilongTang I added the code for unsubscribe in the next commit:

File: test/DynamoCoreWpfTests/ConverterViewModelTests.cs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        }
+
+        protected override void GetLibrariesToPreload(List<string> libraries)
+        {
+            SetupTests();
+            libraries.Add("VMDataBridge.dll");
+            libraries.Add("DSCoreNodes.dll");
+            libraries.Add("DynamoConversions.dll");
+            libraries.Add("DynamoUnits.dll");
+            libraries.Add("CoreNodeModelsWpf.dll");
```
Comment (aparajit-pratap) [inline test/DynamoCoreWpfTests/ConverterViewModelTests.cs:44]: CoreNodeModelsWpf cannot be imported into the VM. I thought I had warned against doing this @RobertGlobant20. Please change.

Comment (aparajit-pratap) [reply test/DynamoCoreWpfTests/ConverterViewModelTests.cs:44]: Also please don't call `SetupTests` inside of `GetLibrariesToPreload`. This method should purely be used to preload libraries.

File: test/DynamoCoreWpfTests/ConverterViewModelTests.cs [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+        public void SetupTests()
+        {
+            var assemblyPath = Assembly.GetExecutingAssembly().Location;
+            var moduleRootFolder = Path.GetDirectoryName(assemblyPath);
+
+            var resolutionPaths = new[]
+            {
+                // The CoreNodeModelsWpf.dll needed is under "nodes" folder.
+                Path.Combine(moduleRootFolder, "nodes")
```
Comment (aparajit-pratap) [inline test/DynamoCoreWpfTests/ConverterViewModelTests.cs:56]: I think this test fixture should be using the `Setup` class test fixture that already adds this resolution path so this should not be needed again here. The `Setup` class is defined in `Setup.cs` in the same `DynamoCoreWpfTests` assembly.

**Review by aparajit-pratap — COMMENTED** [review]
Apologize for the late review but please make the necessary changes as mentioned in my comments and submit a new PR.

File: test/DynamoCoreWpfTests/ConverterViewModelTests.cs (hunk 3 above)
Comment (QilongTang) [reply test/DynamoCoreWpfTests/ConverterViewModelTests.cs:44]: @aparajit-pratap I think you mean overwrite the existing `Start` function from base?

File: test/DynamoCoreWpfTests/ConverterViewModelTests.cs (hunk 4 above)
Comment (QilongTang) [reply test/DynamoCoreWpfTests/ConverterViewModelTests.cs:56]: @aparajit-pratap Reminded me that we are already doing these in the base class

File: test/DynamoCoreWpfTests/ConverterViewModelTests.cs (hunk 3 above)
Comment (RobertGlobant20) [reply test/DynamoCoreWpfTests/ConverterViewModelTests.cs:44]: @aparajit-pratap  
> CoreNodeModelsWpf cannot be imported into the VM. I thought I had warned against doing this @RobertGlobant20. Please change.

Related to this comment, I remember that you told me that I cannot add the dll in this way libraries.Add("CoreNodeModelsWpf.dll"); but I realized that I was testing UI features and if I don't add the "CoreNodeModelsWpf.dll" the next line returns null (then is not possible to get the DynamoConverterControl):

 var dynamoConverterControl = nodeView.grid.ChildrenOfType<DynamoConverterControl>().FirstOrDefault();

If is not a good idea adding the dll in this way, do you have any idea how can I add/load the dll?

cc: @QilongTang

Comment (aparajit-pratap) [reply test/DynamoCoreWpfTests/ConverterViewModelTests.cs:44]: The dll should be loaded by the resolve assembly logic, not this. This list of dll's are for loading them into the Dynamo engine as nodes.

**Pull request merged**